### DISCUSSION TITLE
Test error :Invalid type for argument in function call. Invalid implicit conversion from contract YeildERC20 to contract MockUSDC requested

### DISCUSSION THREAD

**Discussion by guyzilberblum** [Cyfrin/security-and-auditing-full-course-s23#180] [Q&A]
as shown in the title i had this error i solved it by changing the place of the varible 
"""

      this is the wrong one 
      first mockUSDC, and then yeildERC20
      handler = new Handler(handlerStatefulFuzzCatches,mockUSDC,yeildERC20, user);

     this is the right one 
     first yeildERC20 and then mockUSDC
     handler = new Handler(handlerStatefulFuzzCatches,yeildERC20,mockUSDC, user);
""" 

as you can see  I just changed the order my quastion is why in patricks terminal at the youtube course it did not gave him this error although he wrote it like the first order 
also i think that  the order of these 2 may have an impact but im not really sure 
like if one come first then at the above also the one needs to be first 
"""

         yeildERC20 = new YeildERC20();
         mockUSDC = new MockUSDC();
"""

**Comment by EngrPips** [comment]
```js
this is the wrong one 
  first mockUSDC, and then yeildERC20
  handler = new Handler(handlerStatefulFuzzCatches,mockUSDC,yeildERC20, user);

 this is the right one 
 first yeildERC20 and then mockUSDC
 handler = new Handler(handlerStatefulFuzzCatches,yeildERC20,mockUSDC, user);
```

**Reply by EngrPips** [reply]
```js
 yeildERC20 = new YeildERC20();
 mockUSDC = new MockUSDC();
```

**Reply by EngrPips** [reply]
Hello @guyzilberblum, That strictly depends on how you arranged the arguments in the `Handler` constructor. You must have arranged the constructor argument of your `Handler` contract differently than Patrick arranged them.

**Reply by guyzilberblum** [reply]
ok got it thanks

**Reply by EngrPips** [reply]
Anytime, Friend. I am glad you got it.
